### PR TITLE
Allow reseting test framework to same value after setting options

### PR DESCRIPTION
**Pull request by tresat**
Prior to blocking a call to set the test framework on a Test task, check if the value is already finalized to the value and if so; allow it.

Needed to fix an Android Studio issues using Gradle 8.0 milestone 3.

- This also adds a isFinalized method to AbstractProperty to support this use case.
- Also expand and adjust some tests to now pass, and explicitly test the built-in test task and custom Test tasks outside of Test Suites.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/model-core/src/main/java/org/gradle/api/internal/provider/AbstractProperty.java [hunk 1]
```diff
@@ -51,6 +51,15 @@ protected void init(S initialValue) {
...
     }
 
+    /**
+     * A simple getter that checks if this property has been finalized.
+     *
+     * @return {@code true} if this property has been finalized, {@code false} otherwise
+     */
+    public boolean isFinalized() {
+        return state instanceof FinalizedValue;
+    }
```
Comment (big-guy) [inline subprojects/model-core/src/main/java/org/gradle/api/internal/provider/AbstractProperty.java:61]: Can you add a test for this? Maybe PropertySpec or nearby

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy [hunk 2]
```diff
@@ -339,6 +339,33 @@ class TestTaskIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         failure.assertHasCause("The value for task ':test' property 'testFrameworkProperty' is final and cannot be changed any further.")
     }
 
+    def "options can be set prior to setting same test framework for a custom test task"() {
+        ignoreWhenJUnitPlatform()
```
Comment (big-guy) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy:343]: Can you also add the simplest case:

```
test {
  useJUnit { }
  useJUnit { }
}
```

**Review by big-guy — CHANGES_REQUESTED** [review]

**Comment by tresat** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

**Pull request merged**